### PR TITLE
DEV_Petrov

### PR DESCRIPTION
**Pull request by grom4er**
Create Bank application.
Add Model: User, Role.
Service: User, Role.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,4 +5,6 @@ script:
   - mvn clean install
 cache:
   directories:
-    - $HOME/.m2
\ No newline at end of file
+    - $HOME/.m2
+before_install:
+  - chmod +x mvnw
```
Comment (vicire) [inline .travis.yml:10]: Add empty line

File: src/main/java/bank/app/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import lombok.Data;
+
+@Data
+@Entity
+public class Role {
```
Comment (vicire) [inline src/main/java/bank/app/model/Role.java:13]: Rename table as you made with User entity

File: src/main/java/bank/app/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+public interface UserService {
+    User save(User user);
+
+    void delete(Long id);
+
+    User getById(Long id);
+
+    List<User> findAll();
```
Comment (vicire) [inline src/main/java/bank/app/service/UserService.java:13]: we don't need to implement this method. On the other hand, you didn't declare and implement method `update`

File: src/main/java/bank/app/service/impl/RoleServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Override
+    public Role save(Role role) {
+        return repository.save(role);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        return repository.getByRoleName(Role.Roles.valueOf(roleName))
+               .orElseThrow(() -> new UserException((
```
Comment (vicire) [inline src/main/java/bank/app/service/impl/RoleServiceImpl.java:25]:
```suggestion
               .orElseThrow(() -> new NoSuchEntityException((
```
Make your exception more general. That will help you with other entities

File: src/main/java/bank/app/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public void delete(Long id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public User getById(Long id) {
+        return repository.getUserById(id)
+                .orElseThrow(() -> new UserException("Can't find user by id " + id));
```
Comment (vicire) [inline src/main/java/bank/app/service/impl/UserServiceImpl.java:31]: Same here

**Review by vicire — CHANGES_REQUESTED** [review]
Nice work

**Review by vicire — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -83,6 +83,14 @@
                     <includeTestSourceDirectory>true</includeTestSourceDirectory>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>9</source>
```
Comment (boroda4436) [inline pom.xml:90]: maybe it's better to use `jdk.version` property?

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**